docs/docs/themes/shadowing.md

```diff
@@ -0,0 +1,215 @@
+    │   ├── post.js
+    │   ├── posts.js
+    │   ├── seo.js
+    │   └── switch.js
+    ├── gatsby-plugin-theme-ui
+    │   ├── colors.js
+    │   ├── components.js
+    │   ├── index.js
+    │   ├── prism.js
+    │   ├── styles.js
+    │   └── typography.js
+    └── templates
+        ├── post.js
+        └── posts.js
+```
+
+### Customizing the `Bio` component
+
+In this case, the file to shadow is `gatsby-theme-blog/src/components/bio.js`.
+
+Component shadowing relies on a mirrored file structure to determine which component will be rendered. In order to override the Bio component in `gatsby-theme-blog`, create a file named `user-site/src/gatsby-theme-blog/components/bio.js`.
+
+Any file that lives in the `src/gatsby-theme-blog` directory of the user’s site will be used instead of a file with the same name located in the theme’s src directory: `gatsby-theme-blog/src`.
+
+This means that `user-site/src/gatsby-theme-blog/components/bio.js` will be rendered in place of `gatsby-theme-blog/src/components/bio.js`:
+
+```js:title=src/gatsby-theme-blog/components/bio.js
+import React from "react"
+export default () => <h1>My new bio component!</h1>
+```
+
+A successful shadow of the Bio component will result in the following directory tree:
+
+```txt
+user-site
+└── src
+    └── gatsby-theme-blog
+        └── components
+            └── bio.js // highlight-line
+```
+
+## Shadowing parent themes
+
+Some themes, including `gatsby-theme-blog`, install other themes. `gatsby-theme-blog` uses `gatsby-plugin-theme-ui`. If you want to customize any of its implementation you can do so with shadowing as well.
+
+To do so, create a file named `user-site/src/gatsby-theme-blog/gatsby-plugin-theme-ui/index.js`.
+
+```js:title=src/gatsby-theme-blog/gatsby-plugin-theme-ui/index.js
+export default {
+  fontSizes: [12, 14, 16, 24, 32, 48, 64, 96, 128],
+  space: [0, 4, 8, 16, 32, 64, 128, 256],
+  colors: {
+    blue: `blue`,
+    red: `tomato`,
+  },
+}
+```
+
+You’ll result in the following directory tree:
+
+```txt
+user-site
+└── src
+    └── gatsby-theme-blog
+        └── components
+            └── tokens
+                  └──index.js // highlight-line
+```
+
+## Any source file is shadowable
+
+Component shadowing isn’t restricted to React components; you can override any JavaScript, Markdown, MDX, or CSS file in the `src` directory. This gives you fine-grained control of all functionality, content, and styling that a theme provides.
+
+If you wanted to shadow a CSS file in `gatsby-theme-awesome-css` that's found at `src/styles/bio.css` you can do so by creating `user-site/src/gatsby-theme-awesome-css/styles/bio.css`
+
+```js:title=user-site/src/gatsby-theme-awesome-css/styles/bio.css
+.bio {
+  border: 10px solid tomato;
+
+```
+
+The theme's `bio.css` file would then be replaced with your new CSS file.
+
+## Extending shadowed files
+
+In addition to overriding files, you can _extend_ any JavaScript file in the `src` directory.
```

Comment:
Technically you can extend any file type that supports it right? like if there are SASS partials in some .scss file, etc you could import and use them to construct a new classname.

```suggestion
In addition to overriding files, you can _extend_ shadowable files.
```